### PR TITLE
introducing lamport balance monitor for packet verifiers

### PR DESCRIPTION
**Pull request by jeffgrunewald**
allow monitoring the lamport balance of oracles that burn lamports in order to process transactions directly on the solana chain. this will allow the oracles to avoid running too low on a balance to maintain operations without advance notice allowing the service maintainer to top up the account

### CODE REVIEW

**Review by bbalser — APPROVED** [review]

File: solana/src/balance_monitor.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                match handle.await {
+                    Ok(()) => Ok(()),
+                    Err(err) => Err(Error::from(err)),
+                }
+            })
+        }
+    })
+}
+
+async fn run(metric_name: String, solana: Arc<SolanaRpc>, shutdown: triggered::Listener) {
```
Comment (maplant) [inline solana/src/balance_monitor.rs:35]: You should be able to make these generic over `SolanaNetwork` and not have to match on an Option

Comment (maplant) [reply solana/src/balance_monitor.rs:35]: ehhh I guess not

File: solana/src/balance_monitor.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    loop {
+        let shutdown = shutdown.clone();
+
+        tokio::select! {
+            _ = shutdown => {
+                tracing::info!("sol monitor: shutting down");
+                break
+            }
+            _ = trigger.tick() => {
+                let verifier_pubkey = match Keypair::from_bytes(&solana.keypair) {
```
Comment (maplant) [inline solana/src/balance_monitor.rs:47]: why do this every tick?

**Review by bbalser — APPROVED** [review]

**Comment by jeffgrunewald** [comment]
Screenshot 2023-04-24 at 10 39 30 PM
With this change, the `iot_packet_verifier_sol_balance` metric can be seen here being exported

**Review by andymck — APPROVED** [review]

**Review by bbalser — APPROVED** [review]

**Pull request merged**